hawc/apps/epi/api.py

```diff
@@ -87,37 +100,425 @@ def result_heatmap(self, request, pk):
         return Response(export)
 
 
-class StudyPopulation(AssessmentViewset):
+class Criteria(PermCheckerMixin, AssessmentEditViewset):
+    perm_checker_key = "assessment"
+    assessment_filter_args = "assessment"
+    model = models.Criteria
+    serializer_class = serializers.CriteriaSerializer
+
+
+class StudyPopulation(PermCheckerMixin, viewsets.ModelViewSet):
+    perm_checker_key = "study"
     assessment_filter_args = "study__assessment"
     model = models.StudyPopulation
     serializer_class = serializers.StudyPopulationSerializer
 
+    criteria_categories = (
+        ("inclusion_criteria", "I",),
+        ("exclusion_criteria", "E",),
+        ("confounding_criteria", "C",),
+    )
+
+    def process_criteria_association(self, serializer, study_population_id, post_initial_create):
+        # this should probably be rewritten to use the ManyToManyManager on the underlying instance...
+        inserts = []
+        for cc in self.criteria_categories:
+            data_key = cc[0]
+            type_code = cc[1]
+
+            if data_key in self.request.data:
```

Comment:
do in two passes? first find what we need to delete and what we need to insert? do all deleets and all inserts at once?